xarray/core/dataarray.py

```diff
@@ -398,9 +398,22 @@ def __init__(
             if attrs is None and not isinstance(data, PANDAS_TYPES):
                 attrs = getattr(data, "attrs", None)
 
+            def compute_delayed_tuple_elements(tuple_):
+                tuple_ = tuple(
+                    [
+                        elem.compute() if hasattr(elem, "compute") else elem
+                        for elem in tuple_
+                    ]
+                )
+
+                return tuple_
+
+            shape = compute_delayed_tuple_elements(data.shape)
+            coords = compute_delayed_tuple_elements(coords)
```

Comment:
this breaks a lot of the tests (for example, `DataArray([1])` fails). The idea here is to add support for unknown chunk sizes, right? Maybe we should leave that to a different PR? Thoughts, @shoyer, @dcherian?

Reply:
yes we shouldn't do this, it'll compute dask arrays too!

 I don't understand why it's necessary either since `lengths=True` when we call `to_dask_array`: https://docs.dask.org/en/latest/dataframe-api.html#dask.dataframe.DataFrame.to_dask_array

Reply:
is there a way to compute the shape of all columns + the index in one operation without combining them? Combining is not ideal because the columns might have different dtypes, and computing separately means that `raise_if_dask_computes` detects more than 1 compute.